### ISSUE TITLE
libtengine generates bad installation cmake rules

### ISSUE THREAD

**Issue by h6197627** [opencv/opencv#17042]
In OpenCV 4.3.0 built for Android with Tengine support bad Tengine library installation scripts are generated.

1. Configuring for building static OpenCV library, while generating build files it throws:
```
CMake Error: install(EXPORT "OpenCVModules" ...) includes target "opencv_dnn" which requires target "tengine" that is not in any export set.
CMake Error in CMakeLists.txt:
  export called with target "opencv_dnn" which requires target "tengine" that
  is not in any export set.
```
and
```
Error in generation process, project files may be invalid
```
although, despite error it is possible to successfully build OpenCV.

2. It does not respect installation tree structure while installing statically built OpenCV:
Instead of installing into directory where all third party library resides (**sdk/native/3rdparty/libs/***) it puts it into root installation directory. Also it installs include files in the root directory which are actually not needed and are not installed for other 3rd party libraries.

3. While building shared version of OpenCV it also installs includes and static library into root installation directory. For shared OpenCV configuration it is not needed and all other 3rd party static libraries compiled during building process are omitted from installation.

**Comment by asmorkalov** [comment]
@vpisarev Could you look at the issue?

**Comment by liqi-c** [comment]
@h6197627  
There two ways to solve this problem . 
1， you may can build with  -DBUILD_SHARED_LIBS=ON 
2,    you can modify code in  file 3rdparty/libtengine/tengine.cmake 
modify   ‘ SET(Tengine_LIB "tengine" CACHE INTERNAL "")‘
to --> ‘SET(Tengine_LIB "${CMAKE_BINARY_DIR}/lib/${ANDROID_ABI}/libtengine.a" CACHE INTERNAL "")’ 
If you have any questions with tengine , you can @liqi-c  , thank you

**Comment by asmorkalov** [comment]
@liqi-c Could you propose PR with fix?

**Comment by h6197627** [comment]
Hi @liqi-c,
thanks, it fixes cmake error. Although, I think that the right way should be somewhat similar to [DownloadADE.cmake](https://github.com/opencv/opencv/blob/master/modules/gapi/cmake/DownloadADE.cmake) as it probably might solve installation path issues as well

**Comment by liqi-c** [comment]
> @liqi-c Could you propose PR with fix?

ok ， I will  try to PR for fix this problem .

**Comment by asmorkalov** [comment]
The issue also reproducible with cross-compiler for ARM Linux:
```
cmake -DBUILD_SHARED_LIBS=OFF -DWITH_TENGINE=ON -DCMAKE_TOOLCHAIN_FILE=../opencv/platforms/linux/aarch64-gnu.toolchain.cmake -DGCC_COMPILER_VERSION=7 ../opencv
```

**Comment by h6197627** [comment]
Fixed in #17468
